test/unit/PollCreator.js

```diff
@@ -46,27 +60,97 @@ describe("PollCreator", () => {
         before(async () => {
             pollCreator = await (
                 await ethers.getContractFactory("PollCreator")
-            ).deploy(token.address)
+            ).deploy(bondingManagerMock.address)
         })
 
-        it("revert when not enough tokens approved", async () => {
+        it("revert when caller has insufficient stake", async () => {
+            const cost = await pollCreator.POLL_CREATION_COST()
+
+            // pendingStake() == 0 && transcoderTotalStake() == 0
+            await expect(pollCreator.createPoll(hash)).to.be.revertedWith(
+                "PollCreator#createPoll: INSUFFICIENT_STAKE"
+            )
+
+            // pendingStake() > 0 && transcoderTotalStake() == 0
+            bondingManagerMock.pendingStake.returns(cost.sub(1))
+            await expect(pollCreator.createPoll(hash)).to.be.revertedWith(
+                "PollCreator#createPoll: INSUFFICIENT_STAKE"
+            )
+
+            // pendingStake() == 0 && transcoderTotalStake() > 0
+            bondingManagerMock.pendingStake.returns(0)
+            bondingManagerMock.transcoderTotalStake.returns(cost.sub(1))
+            await expect(pollCreator.createPoll(hash)).to.be.revertedWith(
+                "PollCreator#createPoll: INSUFFICIENT_STAKE"
+            )
+
+            // pendingStake() > 0 && transcoderTotalStake() > 0
+            bondingManagerMock.pendingStake.returns(cost.sub(1))
             await expect(pollCreator.createPoll(hash)).to.be.revertedWith(
-                "LivepeerToken transferFrom failed"
+                "PollCreator#createPoll: INSUFFICIENT_STAKE"
             )
         })
 
         it("creates a poll", async () => {
-            await token.setMockBool(
-                functionSig("transferFrom(address,address,uint256)"),
-                true
-            )
+            const cost = await pollCreator.POLL_CREATION_COST()
+
+            // pendingStake() > POLL_CREATION_COST
+            bondingManagerMock.pendingStake.returns(cost.add(1))
+
             const start = await fixture.rpc.getBlockNumberAsync()
             const end = start + POLL_PERIOD + 1 // + 1 because createPoll tx will mine a new block
-            const tx = await pollCreator.createPoll(hash)
-            const receipt = await tx.wait()
-            await expect(tx.hash)
+
+            let tx = await pollCreator.createPoll(hash)
+            let receipt = await tx.wait()
+            await expect(tx)
```

Comment:
#minor
Since the transaction is already mined, using tx.hash in expect will be tad bit faster than passing the whole tx, as [internally](https://github.com/EthWorks/Waffle/blob/master/waffle-chai/src/matchers/misc/transaction.ts#L7-L8) it will add 1 layer of extra promise resolution to get the tx hash. 
Can skip this edit for now.